### PR TITLE
[12_3_X] Change regTrainedWithPS to true and update EGM HLT supercluster regression tags for Run 3 in MC

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

See https://github.com/cms-sw/cmssw/pull/37588/commits

#### PR validation:

test parameters:
  - workflows = 12034.0,11634.0,7.23,159.0,12434.0,12834.0

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Backport of https://github.com/cms-sw/cmssw/pull/37588

### CODE REVIEW

**Comment by tvami** [comment]
test parameters:
  - workflows = 12034.0,11634.0,7.23,159.0,12434.0,12834.0

**Comment by tvami** [comment]
Backport of https://github.com/cms-sw/cmssw/pull/37588

**Comment by tvami** [comment]
assign hlt

File: Configuration/AlCa/python/autoCond.py [hunk 1]
```diff
@@ -34,13 +34,13 @@
     # GlobalTag with fixed snapshot time for Run2 HLT RelVals: customizations to run with fixed L1 Menu
     'run2_hlt_relval'              : '123X_dataRun2_HLT_relval_v2',
     # GlobalTag for Run3 HLT: it points to the online GT
-    'run3_hlt'                     : '123X_dataRun3_HLT_v7',
+    'run3_hlt'                     : '123X_dataRun3_HLT_v5',
```
Comment (missirol) [inline Configuration/AlCa/python/autoCond.py:37]: Why are the Data GTs going backwards?

Comment (tvami) [reply Configuration/AlCa/python/autoCond.py:37]: thanks! I'll fix it in a min

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by francescobrivio** [comment]
+alca

 - Verbatim backport of #37588
 - Differences in tests are mostly in HLT/EGM as expected

**Comment by missirol** [comment]
+hlt
  - related update of HLT utilities for 2018-related recipes in #37626

**Comment by clacaputo** [comment]
+reconstruction

- verbatim backport of #37588

**Pull request merged**